python/interpret_community/dataset/dataset_wrapper.py

```diff
@@ -312,9 +317,12 @@ def string_index(self, columns=None):
         except ImportError:
             return None
         tmp_dataset = self._dataset
-        # Temporarily convert to pandas for easier and uniform string handling
+        # Temporarily convert to pandas for easier and uniform string handling,
+        # only use top sampled rows to limit memory usage for string type test
         if isinstance(self._dataset, np.ndarray):
-            tmp_dataset = pd.DataFrame(self._dataset, dtype=self._dataset.dtype)
+            tmp_dataset = pd.DataFrame(self._dataset[:SAMPLED_STRING_ROWS, :], dtype=self._dataset.dtype)
+        else:
+            tmp_dataset = tmp_dataset.iloc[:SAMPLED_STRING_ROWS]
         categorical_col_names = list(np.array(list(tmp_dataset))[(tmp_dataset.applymap(type) == str).all(0)])
         if categorical_col_names:
             all_columns = tmp_dataset.columns
```

Comment:
Might be better to move all the logic for categorical case into a function.

Reply:
this string_index function contains categorical logic only anyway